src/openfermion/utils/_qubit_tapering_from_stabilizer.py

```diff
@@ -0,0 +1,406 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Tools to reduce the number of terms and taper off qubits
+using stabilizer conditions. Based on ideas of arXiv:1701.08213. """
+
+import numpy
+from openfermion.ops import QubitOperator
+from openfermion.utils import count_qubits
+
+
+class StabilizerError(Exception):
+    """Stabilizer error class."""
+
+    def __init__(self, message):
+        """Throw custom errors connected to stabilizers."""
+        super().__init__(message)
+
+
+def _check_commuting_stabilizers(stabilizer_list, msg, thres=1e-6):
+    """
+    Auxiliary function checking that stabilizers commute.
+
+    If two stabilizers anti-commute their product
+    will have an imaginary coefficient.
+    This function checks the list of stabilizers (QubitOperator)
+    and raises and error if a complex number is found in
+    any of the coefficients.
+    """
+    for stab in stabilizer_list:
+        if abs(numpy.imag(list(stab.terms.values())[0])) >= thres:
+            raise StabilizerError(msg)
+
+
+def _check_stabilizer_linearity(stabilizer_list, msg):
+    """
+    Auxiliary function checking that stabilizers are linearly independent.
+
+    If two stabilizer are linearly dependent the result
+    after some of their products will be the identity.
+    This function checks the list of stabilizers (QubitOperator)
+    and raises an error if the identity is found.
+    """
+    for stab in stabilizer_list:
+        if list(stab.terms.keys())[0] == ():
+            raise StabilizerError(msg)
+
+
+def _fix_single_term(term, position, fixed_op, other_op, stabilizer):

```

Comment:
These "internal" functions still need argument doc strings.  Later somebody maintaining this will find it easier to read if inputs and outputs are specified.  Ditto on all following internal methods.

Reply:
In fact, what is the motivation for these being "_" named?  Presumably these helper functions are likely to be used in further analysis for this line of research.